org/postgresql/test/jdbc2/ArrayTest.java

```diff
@@ -413,6 +413,31 @@ public void testNonStandardDelimiter() throws SQLException
 
         assertTrue(!arrRS.next());
     }
+    public void testEmptyBinaryArray() throws SQLException
+    {
+        
+        PreparedStatement pstmt = conn.prepareStatement("SELECT '{}'::int[]");
+        // force binary
+        ((org.postgresql.PGStatement)pstmt).setPrepareThreshold(-1);
+        
+        ResultSet rs = pstmt.executeQuery();
+        while (rs.next()) {
+            Array array = rs.getArray(1);
+            if (!rs.wasNull())
+            {
+                ResultSet ars  = array.getResultSet();
+                try
+                {
+                    ars.getMetaData().getColumnType(1); // this throws NPE
+                }
+                catch (NullPointerException ex)
+                {
+                    assertTrue("should not get here",true);
```

Comment:
I would prefer more descriptive error messages.
Think how that would look like for the developer who would observe this error message: "should not get here".

Even though test name puts some context, it would make more sense to use `throw new IllegalStateException("ars.getMetaData().getColumnType(1) failed for empty array in binary mode, however it should return ...", ex)`

Note how your `assertTrue` hides original exception, making it hard to identify the root cause if any.